src/NMEA/EngineState.hpp

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright The XCSoar Project
+
+#pragma once
+
+#include "Validity.hpp"
+#include "time/Stamp.hpp"
+#include "Atmosphere/Temperature.hpp"
+
+/** Information about the engine sensor box values,
+ *  temperatures and revolutions per second.
+ *  Update rate of the engine sensor box is 1 Hz.
+*/
+struct EngineState
+{
+  /** Just one of the 8 bit is used as Flag.
+   *  Lowest bit 0b00000001 = 0x1 set to 1, indicates a valid RPM value.
```

Comment:
This is redundant - we already have `revs_per_sec_available`. I think we don't need this field at all.

Reply:
What will happen when there is no sensor box update for more than 3 seconds? Does the RPM reading goes into invalid without the mechanism of the Validity class?

Reply:
For future use, the sensor box will signal a emergency engine shutdown by setting bit 0x1 to 0.

Reply:
Since you chose a 3s timeout, the value will be considered invalid 3 seconds after the most recent value was received from the sensor, and the InfoBox should show "---" instead of that value.

Reply:
> For future use, the sensor box will signal a emergency engine shutdown by setting bit 0x1 to 0.

I don't understand the exact semantics from your sentence, but this still doesn't justify having this field here - because it's a one-way street. It will never be propagated back to the device. NMEAInfo is just stuff from devices/sensors into XCSoar, but never back.

Reply:
> It will never be propagated back to the device.

Does not need to. RPM will go "---" all other sensor box values will be != "---"  => user knows something is not in order with the engine e.g. emergency shut down of engine.